### PR TITLE
Document berkshelf

### PR DESCRIPTION
**Pull request by thommay**
Also break out the Version Constraint syntax into a separate section.

### CODE REVIEW

File: chef_master/source/berkshelf.rst [hunk 1]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
 =====================================================
 `[edit on GitHub] <https://github.com/chef/chef-web-docs/blob/master/chef_master/source/berkshelf.rst>`__
 
-Berkshelf is a dependency manager for certain cookbook workflows that is included in the Chef development kit. Berkshelf stores every version of a cookbook that has ever been installed. By default, the local disk repository for Berkshelf is located at ``~/.berkshelf``, but this may be modified by setting the ``BERKSHELF_PATH`` environment variable.
+.. tag berkshelf_summary
+
+Berkshelf is a dependency manager for certain cookbook workflows that is included in the Chef development kit.
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: Chef Development Kit according to other places we reference it

File: chef_master/source/berkshelf.rst [hunk 2]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+
+.. note:: For new users, we strongly recommend using :doc:`Policyfiles </policyfile>` rather than Berkshelf.
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: 🎉

File: chef_master/source/berkshelf.rst [hunk 3]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+
+.. tag berksfile_summary
+
+A Berksfile describes the set of sources and dependencies needed to use a cookbook. It is used in conjunction with the ``berks`` command.
+
+.. end_tag
+
+Syntax
+-------
+A Berksfile is a ruby file, in which sources, dependencies, and options may be specified. Berksfiles are modelled closely on Bundler's Gemfile. The syntax is as follows:
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: Ruby is capitalized

File: chef_master/source/berkshelf.rst [hunk 4]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+
+.. code-block:: ruby
+
+  source "https://supermarket.chef.io"
+  metadata
+  cookbook "NAME" [, "VERSION_CONSTRAINT"] [, SOURCE_OPTIONS]
+
+
+Metadata Keyword
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: Shouldn't the metadata and source section be swapped here so we follow the order of the Berksfile?

File: chef_master/source/cookbook_repo.rst [hunk 5]
```diff
@@ -116,7 +116,7 @@ This configuration file has the following settings:
         :default => 'kitty kitty'
 
 ``chef_version``
-   A range of chef-client versions that are supported by this cookbook.
+   A range of chef-client versions that are supported by this cookbook. All :ref:`version constraint operators <cookbook_version_constraints>` are applicable to this field.
```
Comment (lamont-granquist) [inline chef_master/source/cookbook_repo.rst:119]: there might be a difference here, though which is that chef_version supports multiple version constraints, while cookbook deps are still single-constraints and we've never fixed that bug?  so `chef_version ">= 12.5.1", "< 14.0"` is legal while `depends ">= 12.5.1", "< 14.0"` is not.

File: chef_master/source/berkshelf.rst [hunk 6]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+
+This ensures that a cookbook named ``my-cookbook`` is installed by berkshelf.
+
+Version constraints are the second parameter:
+
+.. code-block:: ruby
+
+  cookbook "library-cookbook", "~> 0.1.1"
+
+These are identical to the version constraints in a :ref:`cookbook metadata file <cookbook_version_constraints>`.
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: We have a cookbook versions page that might better explain the constraints. It's also huge and probably a bit too complex

File: chef_master/source/berkshelf.rst [hunk 7]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+
+Source options are used to specify the location to acquire a cookbook from, or to place a cookbook in a group. By default, cookbooks are acquired from the default sources, but it's possible to override this on a case by case basis. Often this is used to get a development cookbok from Git, or to use another cookbook in a monorepo.
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: The monorepo term is probably not understood by most people reading this doc

File: chef_master/source/cookbook_repo.rst (hunk 5 above)
Comment (thommay) [reply chef_master/source/cookbook_repo.rst:119]: How hard would it be to fix the bug? I'd rather do that.

File: chef_master/source/berkshelf.rst [hunk 8]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
...
+    cookbook "test-cookbook", path: "test/fixtures/test"
+  end
+
+Groups can also be specified inline:
+
+.. code-block:: ruby
+  
+  cookbook "test-cookbook", path: "test/fixtures/test", group: :test
+
+To exclude a group when using ``berks``, use the ``--except`` flag:
```
Comment (tas50) [inline chef_master/source/berkshelf.rst]: is it except or exclude? The example below has it different

File: chef_master/source/berkshelf.rst (hunk 6 above)
Comment (thommay) [reply chef_master/source/berkshelf.rst]: Hm. Holy cow. We should fix that page, but I think we need something in the metadata page that gives people a quick reference.

Comment (tas50) [reply chef_master/source/berkshelf.rst]: NM I see you put a simpler section in the doc you linked

File: chef_master/source/cookbook_repo.rst (hunk 5 above)
Comment (lamont-granquist) [reply chef_master/source/cookbook_repo.rst:119]: IDK, hasn't bubble up on my list, but yes, I'd vastly prefer to get that fixed.

File: chef_master/source/config_rb_metadata.rst [hunk 9]
```diff
@@ -224,7 +234,7 @@ This configuration file has the following settings:
       name 'cats'
 
 ``ohai_version``
-   A range of chef-client versions that are supported by this cookbook.
+   A range of chef-client versions that are supported by this cookbook. All :ref:`version constraint operators <cookbook_version_constraints>` are applicable to this field.
```
Comment (tas50) [inline chef_master/source/config_rb_metadata.rst]: ohai versions not chef-client versions

**Review by tas50 — CHANGES_REQUESTED** [review]
This is going to be a huge win for first time users. Just fix the minor feedback and ship it

File: chef_master/source/berkshelf.rst (hunk 8 above)
Comment (thommay) [reply chef_master/source/berkshelf.rst]: it is except.

File: chef_master/source/cookbook_repo.rst (hunk 5 above)
Comment (thommay) [reply chef_master/source/cookbook_repo.rst:119]: And actually we don't really doc the multi constraint version on this page, so that's fine.

**Comment by thommay** [comment]
@tas50 all changes made.

**Review by tas50 — APPROVED** [review]

**Review by lamont-granquist — APPROVED** [review]

File: chef_master/source/berkshelf.rst [hunk 10]
```diff
@@ -3,7 +3,181 @@ About Berkshelf
 =====================================================
 `[edit on GitHub] <https://github.com/chef/chef-web-docs/blob/master/chef_master/source/berkshelf.rst>`__
 
-Berkshelf is a dependency manager for certain cookbook workflows that is included in the Chef development kit. Berkshelf stores every version of a cookbook that has ever been installed. By default, the local disk repository for Berkshelf is located at ``~/.berkshelf``, but this may be modified by setting the ``BERKSHELF_PATH`` environment variable.
+.. tag berkshelf_summary
+
+Berkshelf is a dependency manager for certain cookbook workflows that is included in the Chef Development kit.
+
+.. note:: For new users, we strongly recommend using :doc:`Policyfiles </policyfile>` rather than Berkshelf.
```
Comment (davidwrede) [inline chef_master/source/berkshelf.rst:10]: A brief follow-on sentence as to "why" would be helpful here.

**Review by davidwrede — APPROVED** [review]
Everything looks 👍 . Just a minor comment that can be addressed in a different PR.

**Comment by MarkGibbons** [comment]
On Dec 2, 2016, at 3:24 PM, David Wrede <notifications@github.com<mailto:notifications@github.com>> wrote:


@davidwrede approved this pull request.

Everything looks 👍 . Just a minor comment that can be addressed in a different PR.

**Comment by davidwrede** [comment]
gif-keyboard-18413788299948847258

**Pull request merged**